### PR TITLE
Docs: Update the view spec

### PR DESCRIPTION
**Pull request by ajantha-bhat**
some follow-up from https://github.com/apache/iceberg/pull/3188

Other than some comments to add detailed examples, I have handled all. Examples can be added in the follow-up PR.

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
cc: @rdblue , @jzhuge

File: format/view-spec.md [hunk 1]
```diff
@@ -58,16 +55,16 @@ Writers create view metadata files optimistically, assuming that the current met
 
 The view version metadata file has the following fields:
 
-| Required/Optional | Field Name | Description |
-|-------------------|------------|-------------|
-| Required | format-version | An integer version number for the view format. Currently, this must be 1. Implementations must throw an exception if the view's version is higher than the supported version. |
-| Required | location | The view's base location. This is used to determine where to store manifest files and view metadata files. |
```
Comment (ajantha-bhat) [inline format/view-spec.md:64]: I was reading the spec today and these 'manifest files' caught my attention. Then I wanted to see why it is missed in the review. Later found that there are open comments on the same. So, handled most of the comments.

File: format/view-spec.md [hunk 2]
```diff
@@ -32,14 +32,11 @@ Each compute engine stores the metadata of the view in its proprietary format in
...
 * A common metadata format for view metadata, similar to how Iceberg supports a common table format for tables.
-* The view metadata format specification
-  * Includes storage format as well as APIs to write/read the metadata.
-  * Supports versioning of views to track how a view evolved over time.
 
 ## Overview
 
 View metadata storage mirrors how Iceberg table metadata is stored and retrieved. View metadata is maintained in metadata files. All changes to view state create a new view metadata file and completely replace the old metadata using an atomic swap. Like Iceberg tables, this atomic swap is delegated to the metastore that tracks tables and/or views by name. The view metadata file tracks the view schema, custom properties, current and past versions, as well as other metadata.
-Each metadata file is self-sufficient. It contains the history of the last few operations performed on the view and can be used to roll back the view to a previous version.
+Each metadata file is self-sufficient. It contains the history of the last few operations performed on the view. The view can be roll back to a previous version using these version ids.
```
Comment (rdblue) [inline format/view-spec.md:39]: I don't think this needs to change.

File: format/view-spec.md (hunk 1 above)
Comment (rdblue) [reply format/view-spec.md:64]: No need for "the". It is fine to say "where to store view metadata files".

Comment (rdblue) [reply format/view-spec.md:64]: Updated.

**Comment by rdblue** [comment]
Thanks, @ajantha-bhat!

**Pull request merged**